OmniKit/PumpManager/OmnipodPumpManager.swift

```diff
@@ -1364,32 +1364,64 @@ extension OmnipodPumpManager: PumpManager {
                 state.bolusEngageState = .engaging
             })
 
-            var podStatus: StatusResponse
-
-            do {
-                podStatus = try session.getStatus()
-            } catch let error {
-                completion(.failure(PumpManagerError.communication(error as? LocalizedError)))
-                return
-            }
-
             // If pod suspended, resume basal before bolusing
-            if podStatus.deliveryStatus == .suspended {
+            if case .some(.suspended) = self.state.podState?.suspendState {
                 do {
                     let scheduleOffset = self.state.timeZone.scheduleOffset(forDate: Date())
                     let beep = self.confirmationBeeps
-                    podStatus = try session.resumeBasal(schedule: self.state.basalSchedule, scheduleOffset: scheduleOffset, acknowledgementBeep: beep, completionBeep: beep)
+                    _ = try session.resumeBasal(schedule: self.state.basalSchedule, scheduleOffset: scheduleOffset, acknowledgementBeep: beep, completionBeep: beep)
                 } catch let error {
+                    self.log.error("enactBolus: error resuming suspended pod: %s", String(describing: error))
                     completion(.failure(PumpManagerError.communication(error as? LocalizedError)))
                     return
                 }
             }
 
-            guard !podStatus.deliveryStatus.bolusing else {
-                completion(.failure(PumpManagerError.communication(PodCommsError.unfinalizedBolus)))
+            var doGetStatus = false
+            var doFinalizeFinishedDoses = false
+            let preError = self.setStateWithResult({ (state) -> PodCommsError? in
```

Comment:
It looks like this block is not actually updating state; you can just access state, or make a local let of state.

Reply:
This is the same basic logic that is used in enactTempBasal() which doesn't allow for a new TBR to be sent if the pod is suspended or an unfinalized bolus is currently in progress. If enactTempBasal and enactBolus() can both be reworked to use a simpler style for the preError checking, please to go for it -- but I'm not comfortable doing this myself at this time. Additionally I know that it has had months of testing in its current form.

Reply:
You can keep the same style as a function if you want:
```
let preError = {
    if case .some(.suspended) = state.podState?.suspendState {
        self.log.error("enactBolus: failing with suspended pod")
        return .podSuspended
    }
    ...
}()

The setStateWithResult makes it looks like it's updating state, which it's not.

Reply:
Rewrote enactBolus(), enactTempBasal() and setTime() to avoid using setStateWithResult